scripts/Diffraction/isis_powder/hrpd.py

```diff
@@ -60,6 +62,48 @@ def mask_prompt_pulses_if_necessary(self, ws_list):
         for ws in ws_list:
             self._mask_prompt_pulses(ws)
 
+    def should_subtract_empty_inst(self):
+        return self._inst_settings.subtract_empty_inst
+
+    def create_solid_angle_corrections(self, vanadium, run_details):
+        """
+        Creates the solid angle corrections from a vanadium run, only applicable on HRPD otherwise return None
+        :param vanadium: The vanadium used to create this
+        :param run_details: the run details of to use
+        """
+        if not self._inst_settings.do_solid_angle:
+            return None
+        solid_angle = mantid.SolidAngle(InputWorkspace=vanadium)
+        scale = mantid.CreateSingleValuedWorkspace(DataValue='100')
+        correction = mantid.Multiply(LHSWorkspace=solid_angle, RHSWorkspace=scale)
+        eff = mantid.Divide(LHSWorkspace=vanadium, RHSWorkspace=correction)
```

Comment:
As this is a user-facing script, can we break this method up a bit, into separate code blocks, with a comment explaining the code block?